includes/class-sensei-learner.php

```diff
@@ -111,6 +114,63 @@ public function delete_all_user_activity( $user_id = 0 ) {
 		return $dataset_changes;
 	}
 
+	/**
+	 * Remove duplicate progress comments to mitigate duplicate enrollment issue.
+	 *
+	 * Hooked into sensei_log_activity_after
+	 *
+	 * @since 3.7.0
+	 * @access private
+	 *
+	 * @param array $args
+	 */
+	public function remove_duplicate_progress( $args ) {
+		add_action(
+			'shutdown',
+			function() use ( $args ) {
+				global $wpdb;
+
+				// Get progress comments, but the first one, which match the context conditions (they should not exist).
+				// phpcs:disable WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching, WordPress.DB.PreparedSQL.NotPrepared
+				$comment_ids = $wpdb->get_col(
+					$wpdb->prepare(
+						"SELECT comment_ID
+							FROM $wpdb->comments
+							WHERE comment_post_ID = %d
+								AND user_id = %d
+								AND comment_type = %s
+							ORDER BY comment_ID
+							LIMIT 1, 99",
+						$args['post_id'],
+						$args['user_id'],
+						$args['type']
```

Comment:
Seems like it might be possible for these array elements not to exist. Should we check for them first?

Reply:
It seems the `sensei_log_activity` is being called with all expected properties for the `$args` in all places, but I got your point. It's really better to make sure it's there to avoid a dangerous issue to run the `DELETE`. =)

https://github.com/Automattic/sensei/pull/3855/commits/1c8cb8c5474235d6c22976a8f7554b21940f985a